### ISSUE TITLE
Ghost switching after HA update 2022.4

### ISSUE THREAD

**Issue by satyambaba** [AlexxIT/SonoffLAN#742]
Multiple Sonoff devices (Sonoff Basics, Sonoff S30, and Sonoff Minis) are ghost switching (turning on and quickly off, and vice-versa) very frequently after the HA update 2022.4.

**Comment by AlexxIT** [comment]
Try disabling this component and see if there is a problem

**Comment by satyambaba** [comment]
> Try to disable component and check if problem exists

I had rolled back to HA 2022.3 and everything was working as expected. Today, I again updated to HA 2022.4 and it started again.

**Comment by satyambaba** [comment]
> Try to disable component and check if problem exists

Removing Sonoff LAN component in HA solves the issue. Moreover these switching did not create any log in the eWeLink app logs.

**Comment by AlexxIT** [comment]
You can enable debug mode and check logs

**Comment by satyambaba** [comment]
> You can enable debug mode and check logs

I was talking about logs in the eWeLink app (not HA). Whenever we turn on/off any sonoff device in HA, it also creates a log in the eWeLink app. The strange thing in this situation is that the devices are ghosting, however it doesn't create a log in the eWelink app. Now, when I removed the Sonoff LAN component from HA, ghost switching has stopped.

**Comment by AlexxIT** [comment]
So you need to check component logs
https://github.com/AlexxIT/SonoffLAN#component-debug-mode

**Comment by AlexxIT** [comment]
I think you relay loosing electric power and reboot very often.

**Comment by satyambaba** [comment]
> I think your relay loosing electric power and reboot very often.

This is highly unlikely, as non-sonoff devices do not show any sign of this issue. And it is happening all across different rooms across the house to only sonoff devices. For example - Lights on sonoff switch flicker during ghost switching, however at the same time no fluctuation is observed for hue lights in the same room. If it would have been a case of losing power, both kinds of lights should have shown flickering.

**Comment by AlexxIT** [comment]
Try to manually poweroff device (electricity) and check this lines in log:

```
2022-04-11 23:44:40  DEBUG    sonoff_local  1000783864 <= Local3 | {'switch': 'on', 'startup': 'stay', 'pulse': 'off', 'pulseWidth': 500, 'rssi': -42, 'ssid': '...', 'bssid': '...'} | 1
2022-04-11 23:44:43  DEBUG    sonoff_cloud  1000783864 <= Cloud3 | {'action': 'sysmsg', 'deviceid': '1000783864', 'apikey': '...', 'params': {'online': True}}
2022-04-11 23:44:44  DEBUG    sonoff_cloud  1000783864 <= Cloud3 | {'action': 'update', 'deviceid': '1000783864', 'apikey': '...', 'userAgent': 'device', 'd_seq': 1, 'params': {'sledOnline': 'off', 'ssid': '...', 'bssid': '...'}, 'from': 'device'}
2022-04-11 23:44:44  DEBUG    sonoff_cloud  1000783864 <= Cloud3 | {'action': 'update', 'deviceid': '1000783864', 'apikey': '...', 'userAgent': 'device', 'params': {'switch': 'on', 'fwVersion': '3.5.1', 'rssi': -42, 'staMac': '...', 'startup': 'stay', 'init': 1, 'pulse': 'off', 'pulseWidth': 500}, 'from': 'device'}
```

**Comment by satyambaba** [comment]
> Try to manually poweroff device (electricity) and check this lines in log:
> 
> ```
> 2022-04-11 23:44:40  DEBUG    sonoff_local  1000783864 <= Local3 | {'switch': 'on', 'startup': 'stay', 'pulse': 'off', 'pulseWidth': 500, 'rssi': -42, 'ssid': '...', 'bssid': '...'} | 1
> 2022-04-11 23:44:43  DEBUG    sonoff_cloud  1000783864 <= Cloud3 | {'action': 'sysmsg', 'deviceid': '1000783864', 'apikey': '...', 'params': {'online': True}}
> 2022-04-11 23:44:44  DEBUG    sonoff_cloud  1000783864 <= Cloud3 | {'action': 'update', 'deviceid': '1000783864', 'apikey': '...', 'userAgent': 'device', 'd_seq': 1, 'params': {'sledOnline': 'off', 'ssid': '...', 'bssid': '...'}, 'from': 'device'}
> 2022-04-11 23:44:44  DEBUG    sonoff_cloud  1000783864 <= Cloud3 | {'action': 'update', 'deviceid': '1000783864', 'apikey': '...', 'userAgent': 'device', 'params': {'switch': 'on', 'fwVersion': '3.5.1', 'rssi': -42, 'staMac': '...', 'startup': 'stay', 'init': 1, 'pulse': 'off', 'pulseWidth': 500}, 'from': 'device'}
> ```

at 17:59 - Power was manually turned off (electricity)
at 18:01 - Power was manually turned on (electricity)
at 18:05 - Ghost switching happened (off, then immediately on)
```yaml
2022-04-12 17:53:53  DEBUG    main          1000237a55 == Init   | {'uiid': 14, 'extra': {'manufacturer': 'Maker', 'model': 'Sonoff', 'sw_version': 'ITA-GZ1-GL v3.5.0'}, 'params': {'init': 1, 'staMac': '...', 'rssi': -33, 'timers': [{'mId': '...', 'type': 'repeat', 'at': '30 13 * * 1,2,3,4,5,6,0', 'coolkit_timer_type': 'repeat', 'enabled': 1, 'do': {'switch': 'on'}}, {'mId': '...', 'type': 'repeat', 'at': '0 14 * * 1,2,3,4,5,6,0', 'coolkit_timer_type': 'repeat', 'enabled': 1, 'do': {'switch': 'off'}}], 'pulseWidth': 1800000, 'version': 8, 'switch': 'on', 'bindInfos': {'alexa': ['...'], 'gaction': ['...', '...', '...', '...', '...', '...'], 'iftttTriggerCnt': 0, 'smartThings': []}, 'uiActive': 7200, 'startup': 'stay', 'pulse': 'off', 'sledOnline': 'off', 'fwVersion': '3.5.0', 'only_device': {'ota': 'success'}, 'ssid': '...', 'bssid': '...', 'cloud': 'online'}}
2022-04-12 17:53:54  DEBUG    sonoff_local  1000237a55 <= Local1 | {'switch': 'on', 'startup': 'stay', 'pulse': 'off', 'sledOnline': 'off', 'pulseWidth': 1800000, 'rssi': -33, 'ssid': '...', 'bssid': '...'} | 2
2022-04-12 17:54:55  DEBUG    sonoff_cloud  1000237a55 => Cloud4 | {'uiActive': 7200} | 1649766295917
2022-04-12 17:54:56  DEBUG    sonoff_local  1000237a55 <= Local3 | {'switch': 'on', 'startup': 'stay', 'pulse': 'off', 'sledOnline': 'off', 'pulseWidth': 1800000, 'rssi': -33, 'ssid': '...', 'bssid': '...'} | 3
2022-04-12 17:54:56  DEBUG    sonoff_cloud  1000237a55 <= Cloud3 | {'error': 0, 'deviceid': '1000237a55', 'apikey': '...', 'sequence': '1649766295917'}
2022-04-12 17:54:56  DEBUG    sonoff_cloud  1000237a55 => Cloud5 | Force update sequence: 1649766296419
2022-04-12 17:54:56  DEBUG    sonoff_cloud  1000237a55 <= Cloud3 | {'error': 0, 'deviceid': '1000237a55', 'apikey': '...', 'sequence': '1649766296419', 'params': {'init': 1, 'staMac': '...', 'rssi': -33, 'timers': [{'mId': '...', 'type': 'repeat', 'at': '30 13 * * 1,2,3,4,5,6,0', 'coolkit_timer_type': 'repeat', 'enabled': 1, 'do': {'switch': 'on'}}, {'mId': '...', 'type': 'repeat', 'at': '0 14 * * 1,2,3,4,5,6,0', 'coolkit_timer_type': 'repeat', 'enabled': 1, 'do': {'switch': 'off'}}], 'pulseWidth': 1800000, 'version': 8, 'switch': 'on', 'bindInfos': {'alexa': ['...'], 'gaction': ['...', '...', '...', '...', '...', '...'], 'iftttTriggerCnt': 0, 'smartThings': []}, 'uiActive': 7200, 'startup': 'stay', 'pulse': 'off', 'sledOnline': 'off', 'fwVersion': '3.5.0', 'only_device': {'ota': 'success'}, 'ssid': '...', 'bssid': '...'}}
2022-04-12 17:59:13  DEBUG    sonoff_local  1000237a55 => Local4 | {'switch': 'off'} !! Timeout 1
2022-04-12 17:59:13  DEBUG    sonoff_cloud  1000237a55 => Cloud4 | {'switch': 'off'} | 1649766552719
2022-04-12 17:59:18  DEBUG    sonoff_local  1000237a55 => Local4 | Check offline with timeout 20s
2022-04-12 17:59:19  DEBUG    sonoff_cloud  1000237a55 <= Cloud3 | {'error': 504, 'reason': 'Request Timeout', 'deviceid': '1000237a55', 'apikey': '...', 'sequence': '1649766552719'}
2022-04-12 17:59:24  DEBUG    sonoff_cloud  1000237a55 <= Cloud3 | {'action': 'sysmsg', 'deviceid': '1000237a55', 'apikey': '...', 'params': {'online': False}}
2022-04-12 17:59:24  DEBUG    sonoff_local  1000237a55 => Local4 | Skip parallel checks
2022-04-12 17:59:24  DEBUG    sonoff_local  1000237a55 => Local4 | {'cmd': 'info'} !! (ConnectionKey(host='...', port=8081, is_ssl=False, ssl=None, proxy=None, proxy_auth=None, proxy_headers_hash=2583151614444242557), OSError(113, "Connect call failed ('...', 8081)"))
2022-04-12 17:59:38  DEBUG    sonoff_local  1000237a55 => Local4 | Check offline with timeout 40s
2022-04-12 17:59:41  DEBUG    sonoff_local  1000237a55 => Local4 | {'cmd': 'info'} !! (ConnectionKey(host='...', port=8081, is_ssl=False, ssl=None, proxy=None, proxy_auth=None, proxy_headers_hash=2583151614444242557), OSError(113, "Connect call failed ('...', 8081)"))
2022-04-12 18:00:18  DEBUG    sonoff_local  1000237a55 => Local4 | Check offline with timeout 60s
2022-04-12 18:00:21  DEBUG    sonoff_local  1000237a55 => Local4 | {'cmd': 'info'} !! (ConnectionKey(host='...', port=8081, is_ssl=False, ssl=None, proxy=None, proxy_auth=None, proxy_headers_hash=2583151614444242557), OSError(113, "Connect call failed ('...', 8081)"))
2022-04-12 18:00:21  DEBUG    sonoff_local  1000237a55 => Local4 | Device offline
2022-04-12 18:01:22  DEBUG    sonoff_local  1000237a55 <= Local3 | {'switch': 'on', 'startup': 'stay', 'pulse': 'off', 'pulseWidth': 1800000, 'rssi': -37, 'ssid': '...', 'bssid': '...'} | 1
2022-04-12 18:01:25  DEBUG    sonoff_cloud  1000237a55 <= Cloud3 | {'action': 'sysmsg', 'deviceid': '1000237a55', 'apikey': '...', 'params': {'online': True}}
2022-04-12 18:01:26  DEBUG    sonoff_cloud  1000237a55 <= Cloud3 | {'action': 'update', 'deviceid': '1000237a55', 'apikey': '...', 'userAgent': 'device', 'd_seq': 1, 'params': {'sledOnline': 'off', 'ssid': '...', 'bssid': '...'}, 'from': 'device'}
2022-04-12 18:01:26  DEBUG    sonoff_cloud  1000237a55 <= Cloud3 | {'action': 'update', 'deviceid': '1000237a55', 'apikey': '...', 'userAgent': 'device', 'params': {'switch': 'on', 'fwVersion': '3.5.0', 'rssi': -37, 'staMac': '...', 'startup': 'stay', 'init': 1, 'pulse': 'off', 'pulseWidth': 1800000}, 'from': 'device'}
2022-04-12 18:05:50  DEBUG    sonoff_local  1000237a55 <= Local3 | {'switch': 'on', 'startup': 'stay', 'pulse': 'off', 'pulseWidth': 1800000, 'rssi': -29, 'ssid': '...', 'bssid': '...'} | 1
2022-04-12 18:05:53  DEBUG    sonoff_cloud  1000237a55 <= Cloud3 | {'action': 'sysmsg', 'deviceid': '1000237a55', 'apikey': '...', 'params': {'online': True}}
2022-04-12 18:05:53  DEBUG    sonoff_cloud  1000237a55 <= Cloud3 | {'action': 'update', 'deviceid': '1000237a55', 'apikey': '...', 'userAgent': 'device', 'd_seq': 1, 'params': {'sledOnline': 'off', 'ssid': '...', 'bssid': '...'}, 'from': 'device'}
2022-04-12 18:05:54  DEBUG    sonoff_cloud  1000237a55 <= Cloud3 | {'action': 'update', 'deviceid': '1000237a55', 'apikey': '...', 'userAgent': 'device', 'params': {'switch': 'on', 'fwVersion': '3.5.0', 'rssi': -29, 'staMac': '...', 'startup': 'stay', 'init': 1, 'pulse': 'off', 'pulseWidth': 1800000}, 'from': 'device'}
```

**Comment by AlexxIT** [comment]
I still think you're having electrical problems. Check the mains voltage. Maybe it's too low or too high.

**Comment by satyambaba** [comment]
> I still think you're having electrical problems. Check the mains voltage. Maybe it's too low or too high.
> 
> The device is definitely rebooting. It resets internal counters and reconnects to the cloud.

I am happy to report that the issue is resolved after I upgraded to HA OS 2022.4.3. I don't know what was causing the issue, but I checked the logs after upgrading to 2022.4.3 and didn't find any reboots. This rules out the possibility of electrical problems, there might be something odd in 2022.4.0 / 2022.4.1 / 2022.4.2 that was causing this issue.
